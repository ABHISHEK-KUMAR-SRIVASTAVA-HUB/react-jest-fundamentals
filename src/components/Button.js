import React from 'react';

const Button  = ({label}) => {
    return <div data-testid="button" className='button'>hi{label}</div>
}

export default Button;