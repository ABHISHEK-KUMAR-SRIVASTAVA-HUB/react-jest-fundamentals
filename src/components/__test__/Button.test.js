import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render , cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer'

afterEach(cleanup);

it('renders without crashing', () => {
    const div  =  document.createElement('div');
    ReactDOM.render(<Button/>, div);
})

it('renders button correctly 1', () =>{
   const { getByTestId} = render(<Button label='test_label' />);
   expect(getByTestId('button')).toHaveTextContent('test_label');
});

it('renders button correctly 2', () =>{
    const { getByTestId} = render(<Button label='test_label_new' />);
    expect(getByTestId('button')).toHaveTextContent('test_label_new');
});

it('matches snapshot1', () => {
    const tree = renderer.create(<Button label='save1' />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot2', () => {
    const tree = renderer.create(<Button label='save12' />).toJSON();
    expect(tree).toMatchSnapshot();
})