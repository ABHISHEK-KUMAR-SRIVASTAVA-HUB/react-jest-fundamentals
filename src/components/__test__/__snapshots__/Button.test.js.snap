// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot1 1`] = `
<div
  className="button"
  data-testid="button"
>
  hi
  save1
</div>
`;

exports[`matches snapshot2 1`] = `
<div
  className="button"
  data-testid="button"
>
  hi
  save12
</div>
`;
